clients/githubrepo/branches.go

```diff
@@ -349,17 +356,31 @@ func copyNonAdminSettings(src interface{}, dst *clients.BranchProtectionRule) {
 		copyBoolPtr(v.AllowsDeletions, &dst.AllowDeletions)
 		copyBoolPtr(v.AllowsForcePushes, &dst.AllowForcePushes)
 		copyBoolPtr(v.RequiresLinearHistory, &dst.RequireLinearHistory)
-		copyInt32Ptr(v.RequiredApprovingReviewCount, &dst.RequiredPullRequestReviews.RequiredApprovingReviewCount)
-		copyBoolPtr(v.RequiresCodeOwnerReviews, &dst.RequiredPullRequestReviews.RequireCodeOwnerReviews)
 		copyStringSlice(v.RequiredStatusCheckContexts, &dst.CheckRules.Contexts)
 
+		// If RequiredApprovingReviewCount is nil, we let the struct RequiredPullRequestReviews point to nil, as it means
+		// that PRs are not required for making changes at the branch.
+		if v.RequiredApprovingReviewCount != nil {
+			dst.RequiredPullRequestReviews = new(clients.PullRequestReviewRule)
```

Comment:
should we add a similar check for 
```go
if dst.RequiredPullRequestReviews == nil {
```

so we only allocate if it doesn't exist like above? I know the one above has a comment saying this one will do it, but it's more fragile to have an expected call ordering.

Reply:
I think the fix has a typo: I'm seeing this still (double call to new):
```go
		if branchRequiresPrs(v) {
			if dst.RequiredPullRequestReviews == nil {
				dst.RequiredPullRequestReviews = new(clients.PullRequestReviewRule)
			}
			dst.RequiredPullRequestReviews = new(clients.PullRequestReviewRule)
```

Reply:
Oh gosh it's a typo indeed, thanks for spotting that!